nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestProbesObject.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.io.marshallers.json.JsonAssert;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.runtime.test.runner.Deploy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * @since 11.1
+ */
+@Deploy("org.nuxeo.ecm.core.management")
+public class TestProbesObject extends ManagementBaseTest {
+
+    protected static final String RUN_COUNT = "run count";
+
+    protected static final String COUNTS = "counts";
+
+    protected static final String ADMINISTRATIVE_STATUS = "administrativeStatus";
+
+    protected static final String INFOS = "infos";
+
+    protected static final String SUCCESS = "success";
+
+    protected static final String NAME = "name";
+
+    public static final String PATH = "site/management/probes";
+
+    @Test
+    public void testAllProbes() throws IOException {
+        try (CloseableClientResponse response = httpClientRule.get(PATH)) {
+            assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonAssert jAssert = JsonAssert.on(node.toString());
+            jAssert.get("entity-type").isEquals("probes");
+            JsonAssert jProbeArray = jAssert.get("entries");
+
+            JsonAssert jProbe = jProbeArray.get(0);
+            testProbeInfo(jProbe);
+        }
+    }
+
+    @Test
+    public void testProbe() throws IOException {
+        try (CloseableClientResponse response = httpClientRule.get(PATH + "/" + ADMINISTRATIVE_STATUS)) {
+            assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonAssert jAssert = JsonAssert.on(node.toString());
+            testProbeInfo(jAssert);
+        }
+    }
+
+    @Test
+    public void testLaunchProbe() throws IOException {
+        int runCount;
+        try (CloseableClientResponse response = httpClientRule.get(PATH + "/" + ADMINISTRATIVE_STATUS)) {
+            assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            runCount = node.get(COUNTS).get(RUN_COUNT).asInt();
+        }
+        try (CloseableClientResponse response = httpClientRule.post(PATH + "/" + ADMINISTRATIVE_STATUS, null)) {
+            assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            JsonAssert jAssert = JsonAssert.on(node.toString());
+            testProbeInfo(jAssert);
+            assertEquals((runCount + 1), node.get(COUNTS).get(RUN_COUNT).asInt());
```

Comment:
Extra parenthesis around sum.

Reply:
had to make it a long to be compliant with sonar